### ISSUE TITLE
Sprolim.predict_ltr is out of date

### ISSUE THREAD

**Issue by minty-da** [aaronstevenwhite/SemanticProtoRoleLinkingModel#1]
The predict_ltr method in Sprolim seems to assume older versions of the Sprolim and SprolimData class. In particular, I'm pretty sure

- it assumes a Sprolim.data_split field, rather than just Sprolim.data
- it assumes each element of data.iteritems() has a unique_argnums field (workaround: iterate over the inner dictionary, which is indexed by numbers of syntactic arguments)
- it assumes each element of data.iteritems() has a ltr_perms field (I have no workaround for this!)

**Comment by aaronstevenwhite** [comment]
Yes. This should raise a NotImplementedError for now. The other thing it assumes is that the CRF that defines the syntax loss only has up to second-order factors, which is what the EACL 2017 model does, but the current model uses highest possible order factors---e.g. if there are three arguments, there will be a single third-order factor. I'd like to eventually include a switch that lets you switch between the two assumptions---or even, up to k-order factors for any k.
